src/operator/gitea.ts

```diff
@@ -299,6 +327,23 @@ async function upsertRepo(
     )
 }
 
+async function createOrgsAndTeams(orgApi: OrganizationApi, existingOrganizations: Organization[], teamIds: string[]) {
+  await Promise.all(
+    teamIds.map((organizationName) => {
+      return upsertOrganization(orgApi, existingOrganizations, organizationName)
+    }),
+  ).then(() => {
```

Comment:
we usually try to avoid chaining. You could do it by defining an asynchronous function 

async createOrgWithTeam(){
 res = await upsertOrganization()
 teamIds.filter( => )

}